htlcswitch/link_test.go

```diff
@@ -816,7 +930,7 @@ func TestExitNodeAmountPayloadMismatch(t *testing.T) {
 	// per-hop payload for amount to be the incorrect value.  The proper
 	// value of the amount to forward should be the amount that the
 	// receiving node expects to receive.
-	hops[0].FwdInfo.AmountToForward = 1
+	hops[0].FwdInfo.AmountToForward = btcutil.SatoshiPerBitcoin + 1
```

Comment:
For inbound fees, the link was modified to be more relaxed about accepting htlcs. If it overpays the invoice amount that we require, we're okay with that. Otherwise we couldn't accept an inbound fee.

In this test, the final hop amount was manually lowered to 1. But because of the link change, the higher htlc amount was still accepted. The change in this test raises the final hop amount so that the htlc is insufficient and triggers the expected `FinalIncorrectHtlcAmount`